### PR TITLE
Generic mechanism to write performance tests

### PR DESCRIPTION
**Pull request by fgiorgetti**
This PR introduces a generic mechanism for writing performance tests to be run in batches,
based on the provided number of skupper sites to iterate through. 

To write a new performance test, you just have to:
- Implement the `PerformanceTest` interface
- Write a test function that calls `RunPerformanceTest`

The generic mechanism will setup the skupper network(s) and run all performance tests.
Once all tests have been executed for all topologies, a summary will be displayed in the end.

Along with the summary it also displays the generated log files (for each client job) and 
json files with the results for each job execution.

### CODE REVIEW

**Comment by fgiorgetti** [comment]
Here is a sample demonstrating how results will be displayed:

```
2022/05/19 23:45:16 Performance test execution summary
  2022/05/19 23:45:16 Generated log and json files
    2022/05/19 23:45:16 APP              SITES    JOB                              OUTPUT FILE
    2022/05/19 23:45:16 iperf3           1        iperf3-clients-1-size-1g         /home/fgiorget/git/skupper/test/integration/performance/tmp/sites-1/iperf3-clients-1-size-1g.log
    2022/05/19 23:45:16                                                            /home/fgiorget/git/skupper/test/integration/performance/tmp/sites-1/iperf3-clients-1-size-1g.json
    2022/05/19 23:45:16 postgres         1        pgbench-clients-1                /home/fgiorget/git/skupper/test/integration/performance/tmp/sites-1/pgbench-clients-1.log
    2022/05/19 23:45:16                                                            /home/fgiorget/git/skupper/test/integration/performance/tmp/sites-1/pgbench-clients-1.json
  2022/05/19 23:45:16 Throughput summary
    2022/05/19 23:45:16 JOB                                              SITES        CLIENTS      THROUGHPUT (GBITS/S)   LATENCY AVG ()      LATENCY 50%    LATENCY 99%   
    2022/05/19 23:45:16 iperf3-clients-1-size-1g                         1            1            1.21                   0.00                0.00           0.00          
    2022/05/19 23:45:16 JOB                                              SITES        CLIENTS      THROUGHPUT (TPS)       LATENCY AVG (MS)    LATENCY 50%    LATENCY 99%   
    2022/05/19 23:45:16 pgbench-clients-1                                1            1            111.38                 8.98                8.98           8.98          
```

**Review by nluaces — APPROVED** [review]
Nice!

File: test/integration/performance/redis_test.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ENV_REDIS_REQUESTS = "REDIS_NUMBER_REQUESTS"
+	ENV_REDIS_CLIENTS  = "REDIS_PARALLEL_CLIENTS"
+	ENV_REDIS_TIMEOUT  = "REDIS_TIMEOUT"
+	ENV_REDIS_CPU      = "REDIS_CPU"
+	ENV_REDIS_MEMORY   = "REDIS_MEMORY"
+)
```
Comment (ajssmith) [inline test/integration/performance/redis_test.go:34]: Maybe for a future enhancement, add a variable to specify a comma separated list of tests to run. Empty list would run "all"

File: test/integration/performance/redis_test.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	"github.com/skupperproject/skupper/test/utils/base"
+	"github.com/skupperproject/skupper/test/utils/k8s"
+	"gotest.tools/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
```
[the remark this answers is not in the document]
Comment (fgiorgetti) [reply test/integration/performance/redis_test.go:29]: Added REDIS_DATASIZE variable to handle it.

File: test/integration/performance/redis_test.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+		LatencyUnit:    common.LatencyUnitMs,
+	}
+	assert.Assert(t, common.RunPerformanceTest(a))
+}
+
+func (p *RedisTest) App() common.PerformanceApp {
+	return common.PerformanceApp(*p)
+}
+
+func (p *RedisTest) Validate(serverCluster, clientCluster *base.ClusterContext, job common.JobInfo) common.Result {
```
[the remark this answers is not in the document]
Comment (fgiorgetti) [reply test/integration/performance/redis_test.go:81]: True... I have improved the Redis job generation to create a job for each test that can be performed.
If empty, it runs all the tests. The tests to run can be customized via REDIS_TESTS (comma separated
list of test names).

**Review by ajssmith — APPROVED** [review]
Tried locally and across multiple cloud sites. Great stuff.

**Review by grs — APPROVED** [review]

**Pull request merged**